### PR TITLE
Add ability to publish article

### LINKED ISSUE #12: Add ability to publish article
Application Owner should be able to publish draft article.

### PR DESCRIPTION
**Pull request by RasimKhusaenov**
### Summary

- Add ability to go to article page from root
- Add ability to publish article
- Remove redundant `required` in article form
- Remove useless `successfully` in messages

### How it works

image
image
image

### Test plan

List of steps to manually test introduced functionality:

- Log in as admin
- Go to http://localhost:3000/articles/#{article.id}
- Click "Publish"
- You are awesome!

### Deploy notes

N/A

### Checklist:

- [ ] I have performed a self-review of my own code
- [x] I have added tests for new functionality
- [x] My pull request is shorter than 500 lines

### References
- Resolves #12 Issue

### CODE REVIEW

File: app/controllers/articles_controller.rb [hunk 1]
```diff
@@ -16,13 +16,19 @@ def create
     respond_with article, action: :index
   end
 
+  def edit; end
```
Comment (RamilyaNigmatullina) [inline app/controllers/articles_controller.rb:19]: Do we need this method? I do not see it in `routes`.

File: app/controllers/articles_controller.rb [hunk 2]
```diff
@@ -16,13 +16,19 @@ def create
...
+
+  def update
```
Comment (RamilyaNigmatullina) [inline app/controllers/articles_controller.rb:21]: What do you think about moving this logic to `Articles::PublicationsController#create`?

File: app/controllers/articles_controller.rb [hunk 3]
```diff
@@ -23,6 +23,6 @@ def authorize_resource!
   end
 
   def article_params
-    params.require(:article).permit(:title, :content)
+    params.require(:article).permit(:title, :content, :published)
```
Comment (RamilyaNigmatullina) [inline app/controllers/articles_controller.rb:26]: Should we revert these changes?

File: app/controllers/articles/publications_controller.rb [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+    expose :article, scope: -> { policy_scope(Article) }
+
+    def create
+      if article.update(published: true)
+        redirect_to root_path, notice: I18n.t("flash.articles.publish.success")
+      else
+        redirect_to article_path(article), notice: I18n.t("helpers.failure")
+      end
+    end
```
Comment (RamilyaNigmatullina) [inline app/controllers/articles/publications_controller.rb:13]: What do you think if we user `respond_with`?
```suggestion
    def create
      article.update(published: true)

      respond_with article, location: root_path
    end
```

```yml
# config/locales/en.yml

en:
  flash:
    articles:
      create:
        success: "Article saved!"
      publications:
        create:
          success: "You've published article!"
```

**Review by RamilyaNigmatullina — APPROVED** [review]

File: config/routes.rb [hunk 5]
```diff
@@ -1,7 +1,10 @@
 Rails.application.routes.draw do
   root to: "articles#index"
 
-  resources :articles, only: %i[new create index show]
+  resources :articles, only: %i[new create index show] do
+    resources :publications, only: %i[create], controller: "articles/publications"
```
Comment (RamilyaNigmatullina) [inline config/routes.rb:5]:
```suggestion
    resources :publications, only: %i[create], module: :articles
```

File: app/views/articles/show.html.slim [hunk 6]
```diff
@@ -1,3 +1,6 @@
 h1 #{article.title}
 p
   | #{article.content}
+
+- if current_user&.administrative_role?
```
Comment (RamilyaNigmatullina) [inline app/views/articles/show.html.slim:5]: Do we need to show this button only if article is unpublished?

Comment (RamilyaNigmatullina) [inline app/views/articles/show.html.slim:5]: What do you think about moving this check to policy? And check if user is authorized to publish the article in the controller before executing create method there?

Comment (RasimKhusaenov) [reply app/views/articles/show.html.slim:5]: Yes, but I'm not sure how I will implement the "back to draft" button 😅

Comment (RamilyaNigmatullina) [reply app/views/articles/show.html.slim:5]: @RasimKhusaenov What is the difficulty?  I can help if needed.  You need to implement action destroy in the Publications controller and a button with action destroy and text "Back to draft".

Comment (RasimKhusaenov) [reply app/views/articles/show.html.slim:5]: These will be different buttons on the same page that will be displayed depending on the status of the article, right?

Comment (RamilyaNigmatullina) [reply app/views/articles/show.html.slim:5]: @RasimKhusaenov Yes that's right.

File: app/policies/article_policy.rb [hunk 7]
```diff
@@ -10,4 +10,8 @@ def resolve
   def create?
     user&.administrative_role?
   end
+
+  def update?
```
Comment (RamilyaNigmatullina) [inline app/policies/article_policy.rb:14]: Do we need this method?

**Review by RamilyaNigmatullina — APPROVED** [review]

File: app/policies/article_policy.rb (hunk 7 above)
Comment (RasimKhusaenov) [reply app/policies/article_policy.rb:14]: No 🥺
